### PR TITLE
Adding a scatter chart

### PR DESCRIPTION
**Pull request by cawasser**

### CODE REVIEW

File: src/cljs/vanilla/widget_defs.cljs [hunk 1]
```diff
@@ -177,4 +177,16 @@
...
                              :viz/style-name   "widget"
                              :viz/tooltip      {:followPointer true}
+                             :viz/icon         "timeline"}}
+
+              {:name        :scatter-widget
+               :basis       :chart
+               :type        :scatter-chart
+               :data-source :scatter-service-data
+               :options     {:viz/title        "Scatter"
+                             :viz/banner-color "purple"
```
Comment (cawasser) [inline src/cljs/vanilla/widget_defs.cljs:199]: let's also make the text "white" using :viz/banner-text-color

File: src/cljs/vanilla/widgets/scatter_chart.cljs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                         :allowDecimals (get options :viz/x-allowDecimals false)}
+                 :startOnTick true
+                 :endOnTick true
+                 :showLastLabel true}
+
+
+   :yAxis       {:title {:text          (get-in data [:data :src/y-title ] "y-axis")
+                         :allowDecimals (get options :viz/y-allowDecimals false)}}
+
+   :legend      {:align         "left"
```
Comment (cawasser) [inline src/cljs/vanilla/widgets/scatter_chart.cljs]: the legend seems to draw in the middle of the graph. let's play with these parameters get it somewhere the graph and tooltips don't cover

**Comment by cawasser** [comment]
also merge the latest master onto this branch so the conflicts get resolved before we try to merge the PR

**Comment by neeldave** [comment]
Hi Chris,
I changed the banner color from purple to white, and for the second change to change the parameter, I can not see the graph/banner inside of scatter plot. I changed the data-source name in widget-defs.cljs from scatter-service-data to scatter-service; however, I still don't see any graph/data on webpage.
Is there something that I am missing that I can not see it?
I have pushed this updated scatter branch.
Thank you,Neel
    On Monday, November 25, 2019, 10:12:51 AM EST, Chris Wasser <notifications@github.com> wrote:  
 
 
also merge the latest master onto this branch so the conflicts get resolved before we try to merge the PR

**Comment by neeldave** [comment]
The purple banner color was good, but white text on it would be more readable. 

Later today I’ll get the code and look at the legend and see if it renders okay. I’ll email you then

C

**Comment by neeldave** [comment]
Ok I will edit the banner. 
Thank you very much Chris


Sent from Yahoo Mail for iPhone


On Monday, December 2, 2019, 3:02 PM, Chris Wasser <cawasser@mac.com> wrote:

The purple banner color was good, but white text on it would be more readable. 
Later today I’ll get the code and look at the legend and see if it renders okay. I’ll email you then
C
Sent from my iPhone

On Dec 2, 2019, at 2:59 PM, Neel Dave <neelkdave@yahoo.com> wrote:



﻿ Hi Chris,
I changed the banner color from purple to white, and for the second change to change the parameter, I can not see the graph/banner inside of scatter plot. I changed the data-source name in widget-defs.cljs from scatter-service-data to scatter-service; however, I still don't see any graph/data on webpage.
Is there something that I am missing that I can not see it?
I have pushed this updated scatter branch.
Thank you,Neel
    On Monday, November 25, 2019, 10:12:51 AM EST, Chris Wasser <notifications@github.com> wrote:  
 
 
also merge the latest master onto this branch so the conflicts get resolved before we try to merge the PR

**Comment by cawasser** [comment]
corrections made and merged with master to clear any conflicts. 0.2.2-SNAPSHOT

**Pull request merged**